packages/core/helper-plugin/src/components/tests/ConfirmDialog.test.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import { lightTheme, ThemeProvider } from '@strapi/design-system';
+import { render as renderRTL, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { IntlProvider } from 'react-intl';
+
+import { ConfirmDialog, ConfirmDialogProps } from '../ConfirmDialog';
+
+describe('ConfirmDialog', () => {
+  const render = (props?: Partial<ConfirmDialogProps>) => ({
+    ...renderRTL(
+      <ConfirmDialog
+        isOpen
+        onConfirm={jest.fn()}
+        onToggleDialog={jest.fn()}
+        bodyText={{
+          id: 'app.components',
+          defaultMessage: 'Are you sure you want to unpublish it?',
+        }}
+        title={{ id: 'app.components.ConfirmDialog.title', defaultMessage: 'Confirmation' }}
+        rightButtonText={{ id: 'global.save', defaultMessage: 'Save' }}
+        {...props}
+      />,
+      {
+        wrapper: ({ children }: { children: React.JSX.Element }) => (
+          <ThemeProvider theme={lightTheme}>
+            <IntlProvider locale="en" messages={{}} defaultLocale="en">
+              {children}
+            </IntlProvider>
+          </ThemeProvider>
+        ),
+      }
+    ),
+    user: userEvent.setup(),
+  });
+
+  it('renders with given props and displays open dialog', async () => {
+    render();
+
+    await waitFor(() => {
+      expect(screen.getByText('Are you sure you want to unpublish it?')).toBeInTheDocument();
+    });
+
+    expect(screen.getByText('Confirmation')).toBeInTheDocument();
+
+    expect(screen.getByText('Save')).toBeInTheDocument();
+  });
+
+  it('verifies dialog is hidden when isOpen is false', () => {
+    render({ isOpen: false });
+
+    expect(screen.queryByText('Confirmation')).not.toBeInTheDocument();
+  });
+
+  it('triggers onConfirm when "Save" button is clicked', async () => {
+    const onConfirm = jest.fn();
+    const { user } = render({ onConfirm });
+
+    await user.click(screen.getByRole('button', { name: 'Save' }));
+
+    expect(onConfirm).toBeCalled();
+  });
+
+  it('triggers onToggleDialog when "Cancel" button is clicked', async () => {
+    const onToggleDialog = jest.fn();
+    const { user } = render({ onToggleDialog });
+
+    await user.click(screen.getByRole('button', { name: 'Cancel' }));
+
+    expect(onToggleDialog).toBeCalled();
+  });
+
+  it('renders component parts and matches text', async () => {
+    const { rerender } = render();
+    rerender(
+      <ConfirmDialog.Root isOpen onToggleDialog={jest.fn()} onConfirm={jest.fn()}>
+        <ConfirmDialog.Body>Are you sure you want to unpublish it?</ConfirmDialog.Body>
+      </ConfirmDialog.Root>
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText('Confirm')).toBeInTheDocument();
```

Comment:
From where this confirm text appears in the component?

Reply:
Confirm is the default text on the confirmation button. I did not see a need to test for the unpublish text again, because that is what is passed to the Body component from <ConfirmDialog> anyway. However, because in the first test cases had altered the confirm button with "Save" text instead of default Confirm, i figured all i needed to do was just test for Confirm being on the button.  

I will make all the suggested changes.